docs/docker.rst

```diff
@@ -0,0 +1,142 @@
+
+    INSPIRE is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+
+    In applying this licence, CERN does not waive the privileges and immunities
+    granted to it by virtue of its status as an Intergovernmental Organization
+    or submit itself to any jurisdiction.
+
+
+Docker (Linux)
+==============
+
+Docker is an application that makes it simple and easy to run processes in a container,
+which are like virtual machines, more resource-friendly, and more dependent
+on the host operating system. For a detailed introduction to the different components of a Docker
+container.
+
+Useful `Tutorial
+<https://www.digitalocean.com/community/tutorials/the-docker-ecosystem-an-introduction-to-common-components>`_.
+
+
+Inspire and Docker
+##################
+
+Get the latest Docker setup.
+
+.. note:: If you are using Mac, please build a simple box with ``docker-engine`` above ``1.10`` and
+         ``docker-compose`` above ``1.6.0``.
+
+Make sure you can run docker without ``sudo``.
+
+- ``id $USER``
+
+  If you are not in the ``docker`` group, run the following command and then restart ``docker``. If this doesn't work, just restart your machine :)
+-  ``sudo usermod -a -G docker $USER``
+
+Get docker-compose:
+
+.. code-block:: console
+
+ sudo pip install docker-compose
+
+- Add ``DOCKER_DATA`` env variable in your ``.bashrc`` or ``.zshrc``. In this directory you will have all the persistent data between Docker runs.
+
+.. code-block:: console
+
+ export DOCKER_DATA=~/inspirehep_docker_data/
+
+By default the virtualenv and everything else will be kept on ``/tmp`` and they will be available only until the next reboot.
+
+- Install a host persistent venv and build assets
+
+.. code-block:: console
+
+ docker-compose pull
+ docker-compose -f docker-compose.deps.yml run --rm pip
+ docker-compose -f docker-compose.deps.yml run --rm assets
+
+- Run the service locally
+
+.. code-block:: console
+
+ docker-compose up
+
+Go to ``localhost:5000``
+
+- Populate database
+
+.. code-block:: console
+
+ docker-compose run --rm web scripts/recreate_records
+
+- Run tests in an **isolated** environment.
+
+  You can choose one of the following tests types:
+
+  - unit
+  - disambiguation
+  - workflows
+  - integration
+  - acceptance
+
+.. code-block:: console
+
+ docker-compose -f docker-compose.test.yml run --rm <tests type>
+ docker-compose -f docker-compose.test.yml kill
+ docker-compose -f docker-compose.test.yml rm
+
+**The kill command is mandatory, otherwise you will use the test DB instead of the correct one in dev.**
+
+I don't know how to kill all the other services just after the run command exited.
```

Comment:
Instead of this phrase, add these tips:

In order to fully cleanup the docker environment:

```shell
# cleanup all the containers
docker rm $(docker ps -qa)

# cleanup all the images
docker rmi $(docker images -q)

# cleanup the virtualenv (careful, if docker_data is set to something you care about, it will be removed)
sudo rm -rf "${DOCKER_DATA?DOCKER_DATA was not set, ignoring}"
```